parser.rb

```diff
@@ -0,0 +1,43 @@
+class TextParser
+	attr_reader :text, :word_counts
+	def initialize(file)
+		@text = IO.read(file)
+		@word_counts = {}
+	end
+
+	def remove_newlines
+		text.gsub(/\s+/, ' ').strip
+	end
+
+	def parsed_text
+		remove_newlines.split(" ").map do |string|
+			string.gsub(/[^a-zA-Z0-9'-]/i, '').downcase
+		end
+	end
+
+	def count_words
+		parsed_text.each do |word|
+			if word_counts.key?(word)
+				word_counts[word] += 1
+			else
+				word_counts[word] = 1
+			end
+		end
+	end
+
+	def sorted_counts
+		word_counts.sort_by {|_key, value| -value}
```

Comment:
I bet there are people that wouldn't think this is readable, but I love the simplicity of it.

It would be a slight improvement to give `_key` and `value` more intention-revealing names. `_word` and `count`, I suppose.